PROCESSES.md

```diff
@@ -129,32 +129,114 @@ project itself:
-  then displays final instructions for pushing to nuget.org and
-  updating the docs.
+The last three of these are typically performed by a Kokoro release
+job.
+
+### Detailed steps
+
+**Create the release PR**
+
+1. Edit the API catalog (`apis/apis.json`). Find the package you
+want to release (by ID) and edit the version key to the new release
+version. Usually this will be a bump of prerelease (e.g.
+1.0.0-beta03 to 1.0.0-beta04) or a minor version bump (e.g. 1.3.0 to
+1.4.0). You may update any dependencies at the same time.
+
+2. Run `generateprojects.sh` from the root directory. This should
+indicate that it has updated the project file for the package you're
+releasing.
+
+3. If this is the first release of a package, or it's been updated
+from beta to GA, update `README.md` and `docs/root/index.md`
+accordingly.
+
+4. Update the version history for the package (in
+`apis/{package_id}/docs/history.md`). There is a script in the root
+directory to help with this: run `prepare-release.sh` passing in the
+package ID, which will perform version comparisons and perform an
+initial edit on the file, but you'll need to edit the file manually
+afterwards to make the history as useful as possible. (The tool is
+very new, and we hope to reduce the manual edit requirements over
+time.)
+
+5. Commit the changes. The first line of the commit message should
+be "Release {package_id} version {version}". Optionally, add release
+notes as the rest of the commit message, potentially by copying from
+the `history.md` file you've just edited. ([Sample
+commit](https://github.com/googleapis/google-cloud-dotnet/commit/3b580a6a0e8248daec4c84f6a45d0e07c094013d))
+
+6. Create a pull request for the commit, and get it reviewed.
+
+Sample session when releasing Google.Cloud.Speech.V1:
+
+```text
+$ git checkout -b release-speech
+$ pico apis/apis.json
+$ ./generateprojects.sh
+$ ./prepare-release.sh Google.Cloud.Speech.V1
+$ pico apis/Google.Cloud.Speech.V1/docs/history.md
+$ git commit -a
+$ git push
+```
+
+**Tagging the release**
+
+Prerequisite: the PR above has been reviewed and merged into the
+master branch.
+
+1. Checkout the master branch and pull the latest code, which should
+now include the changes you've made.
+
+2. Run `tagreleases.sh` in the root directory, specifying a github
+access token. This will ask you to confirm that you want to create a
+release.
+
+Sample session:
+
+```text
+$ git checkout master
+$ git pull upstream master
+$ ./tagreleases.sh your_access_token_here
+```
 
 Note that `tagreleases.sh` checks that there are no project
 references from APIs being released now to APIs that *aren't* being
 released. Without this check, it's possible for a released version
 to depend on unreleased changes.
+
+**Building and publishing the release**
+
+On a Google corp machine, trigger a Kokoro release build. (This
+can only be performed by Googlers.)
+
+If you want to release anything other than the current head of
```

Comment:
Aint this what we usually want? The way is written it seems like we usually release master's current head. I tend to always specify the commitish (tagreleases.sh always "returns" the commitish) just in case someone has merged something in between. I think it's safer as general instructions to say "always specify the commitish".

Reply:
I almost always trigger it immediately after running tagreleases.sh, and just let it go with HEAD. It's never caused a problem :) I think it's easier to let that be the default approach - especially as if there are multiple people within in the repo, they're likely to be us!